CMakeLists.txt

```diff
@@ -64,10 +64,15 @@ function(packageProject)
     )
   endif()
 
+  set(wbpvf_extra_args "")
+  if(PROJECT_ARCH_INDEPENDENT)
+    set(wbpvf_extra_args ARCH_INDEPENDENT)
```

Comment:
Actually, since this is [advertised](https://cmake.org/cmake/help/latest/module/CMakePackageConfigHelpers.html#command:write_basic_package_version_file) as a feature mainly for header only libraries, it would be safe to detect the target type and default to using this.

Reply:
Yeah the only situation where I can imagine that this goes wrong is if a library ships completely different headers for difference architectures. But afaik that would be exceptionally rare.

Reply:
Yes, I think the general use-case is not that, so providing a sane default helps here.